toolchain/parse/context.h

```diff
@@ -107,8 +107,7 @@ class Context {
                                kind != NodeKind::InvalidParseStart &&
                                kind != NodeKind::InvalidParseSubtree),
                  "{0} nodes must always have an error", kind);
-    tree_->node_impls_.push_back(
-        {.kind = kind, .has_error = has_error, .token = token});
+    tree_->node_impls_.emplace_back(kind, has_error, token);
```

Comment:
It's (sadly) important to use `push_back` here. Without that, this is a measurable net regression on parsing on M1.

```suggestion
    tree_->node_impls_.push_back(Tree::NodeImpl(kind, has_error, token));
```